### PR TITLE
upgrade react and various fixes

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

**Comment by thepiwo** [comment]
There are still minor issues with the version, will have to check and fix them

File: app/background-tracking/actions.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import BreakoutApi from "breakout-api-client";
 import {BASE_URL, CLIENT_NAME, CLIENT_SECRET, DEBUG} from "../config/secrets";
 import {withAccessToken} from "../utils/utils";
 import {ToastAndroid} from "react-native";
+import _ from "lodash";
```
Comment (nerdsupremacist) [inline app/background-tracking/actions.js:5]: why did you need this?

File: app/components/posting-list.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ export default class PostingList extends React.PureComponent {
             <FlatList ListHeaderComponent={errorHeader}
                       ListFooterComponent={errorFooter}
                       data={props.postings}
-                      keyExtractor={item => item.id}
+                      keyExtractor={item => item.id.toString()}
```
Comment (nerdsupremacist) [inline app/components/posting-list.js:29]: Can't the key extractor handle number values? Or are we handling something like `null` or `undefined`

File: app/screens/all-teams/screen.js [hunk 3]
```diff
@@ -69,6 +70,9 @@ class AllTeams extends React.PureComponent {
                 fontWeight: 'bold',
                 padding: 10
             },
+            imageContainer: {
```
Comment (nerdsupremacist) [inline app/screens/all-teams/screen.js:73]: niiiice ❤️

File: app/screens/postings/screen.js [hunk 4]
```diff
@@ -15,6 +15,10 @@ class PostingListScreen extends Component {
     render() {
         return <PostingList {...this.props}/>
     }
+
+    componentDidMount() {
+        this.props.fetchNewPostings();
```
Comment (nerdsupremacist) [inline app/screens/postings/screen.js]: soooo we weren't loading them before? or were we loading them in another component?

File: app/config/secrets.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const BASE_URL = "specify-url-here";
```
Comment (nerdsupremacist) [inline app/config/secrets.js:1]: Where were these before?

File: app/components/posting-list.js (hunk 2 above)
Comment (thepiwo) [reply app/components/posting-list.js:29]: I'm unsure about null or undefined. React changed and now requires this to be string. Having this as number did lead to error messages in the log.

File: app/screens/postings/screen.js (hunk 4 above)
Comment (thepiwo) [reply app/screens/postings/screen.js]: somehow they were implicitly loaded, couldn't figure out, eventually this was a bug before

File: app/config/secrets.js (hunk 5 above)
Comment (thepiwo) [reply app/config/secrets.js:1]: The file was in the same place, but not in git. This did lead to me always reverse engineering what is should look like...

File: app/background-tracking/actions.js (hunk 1 above)
Comment (thepiwo) [reply app/background-tracking/actions.js:5]: as _ is used but not imported. I'm unsure how this did work before, eventually there was some implicit import happening or the bundling did work in other ways so it was already defined globally.

**Comment by thepiwo** [comment]
I do still have to follow up on some issue where the login page is corrupted and not all functionalities have been tested by me yet.

File: android/app/build.gradle [hunk 6]
```diff
@@ -132,7 +132,6 @@ android {
         release {
             minifyEnabled enableProguardInReleaseBuilds
             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
-            signingConfig signingConfigs.release
```
Comment (florianschmidt1994) [inline android/app/build.gradle]: Does removing this have any implications? Did we test a release yet?

File: android/app/src/main/java/org/break_out/breakout/MainApplication.java [hunk 7]
```diff
@@ -32,12 +29,12 @@ public boolean getUseDeveloperSupport() {
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
           new MainReactPackage(),
-            new RNSentryPackage(MainApplication.this),
-            new ReactNativeLocalizationPackage(),
+            new ReactVideoPackage(),
```
Comment (florianschmidt1994) [inline android/app/src/main/java/org/break_out/breakout/MainApplication.java:34]: While we're at it, fix identation?

File: app/screens/postings/screen.js (hunk 4 above)
Comment (florianschmidt1994) [reply app/screens/postings/screen.js]: This screen is just a dumb container for a PostingList. 

PostingList will call `nextPage()` when it reaches the bottom and `onRefresh()` when you're pulling from the top. This mechanism is responsible for fetching postings and should work as intended, so I don't think we need this here.

The only thing I could image is that there might be a blank screen on the initial render (after downloading the app, with no postings in store yet), but still then the logic should be added to the PostingList

**Review by florianschmidt1994 — COMMENTED** [review]
- Does changing the package from com.breakout to org.break_out.* have any implications for releasing stuff?
- Should we add / enforce a certain code style? This adds a lot of reformatting diffs, which makes it kind of confusing
- When loading the global map and clicking a team name, it does not redirect me to the team profile. IDK if this has ever worked before tbh, but if it did, it's broken now.  
- Login works for me, tested with an account that has had a team 2018

File: app/screens/postings/screen.js (hunk 4 above)
Comment (thepiwo) [reply app/screens/postings/screen.js]: will have a look into this, just noticed the blank screen after installing the app

File: android/app/build.gradle (hunk 6 above)
Comment (thepiwo) [reply android/app/build.gradle]: good question; having a beta release is on my todo list; will check

File: android/app/src/main/java/org/break_out/breakout/MainApplication.java (hunk 7 above)
Comment (thepiwo) [reply android/app/src/main/java/org/break_out/breakout/MainApplication.java:34]: this is auto format for some reason; will try again

File: app/components/posting-list.js [hunk 8]
```diff
@@ -13,6 +13,11 @@ export default class PostingList extends React.PureComponent {
         return <Posting addLike={this.props.addLike} {...(row.item)} />;
     }
 
+    componentDidMount() {
+        if (this.props.postings && this.props.postings.length === 0) {
```
Comment (nerdsupremacist) [inline app/components/posting-list.js:17]: so postings starts with `[]`

**Review by nerdsupremacist — APPROVED** [review]
